shared/actions/platform-specific/push.native.tsx

```diff
@@ -347,43 +350,45 @@ function* _checkPermissions(action: ConfigGen.MobileAppStatePayload | null) {
   }
 }
 
-const getStartupDetailsFromInitialPush = (): Promise<
-  | null
-  | {
-      startupFollowUser: string
+function* getStartupDetailsFromInitialPush() {
+  const {push, pushTimeout}: {push: PushGen.NotificationPayload; pushTimeout: boolean} = yield Saga.race({
+    push: Saga.callPromise(isAndroid ? getInitialPushAndroid : getInitialPushiOS),
+    pushTimeout: Saga.delay(10),
```

Comment:
Is there a chance that this will delay startup? I think we're trying to prevent any significantly blocking activity there.

Reply:
It used to delay startup. Now it can only delay startup at most 10ms. This is how you timeout a promise with redux